var/spack/repos/builtin/packages/py-slepc4py/package.py

```diff
@@ -36,10 +37,12 @@ class PySlepc4py(PythonPackage):
     patch("ldshared.patch", when="@:99")
     patch("ldshared-dev.patch", when="@main")
```

Comment:
Out of curiosity, where are these patches needed, and is there any way to solve the issue in slepc/petsc main?

Reply:
I think the code attempts to handle certain corner cases [perhaps with intel compilers?]. And this breaks with spack - as it changes compilers underneath - hence this fix to skip that code during spack build.

@dalcinl I don' think this change will affect other usages - so perhaps ok to add to petsc/slepc mainline?

Reply:
I feel bad hardcoding things in petsc/slepc just for Spack (although maybe Spack is one of the more popular ways of installing things). I do frequently see patches specific to Homebrew, but those also break things for Spack. Maybe a more general solution that checks if the compiler is a symlink or something.

Reply:
I don't have strong opinions about this. Most of these hideous hacks were done just to workaround limitations of Python's distutils.

Reply:
With the change in file names [that was reflected in one of the changes in this PR] - I think we'll need another patch file added to spack for 3.19. So if we can add this to petsc/slepc mainline - then we can avoid adding additional copy of this patch to spack.

I'll create a MR for petsc